### PR TITLE
Refactor out mocking of tmpfile()

### PR DESCRIPTION
**Pull request by sanmai**
The reason I'm so keen to remove these mocks not only because they're kind of surprising in a bad way, but because it is possible to call these function without considering local namespace for a slight performance advantage, [around 4.5% per call](http://veewee.github.io/blog/optimizing-php-performance-by-fq-function-calls/). 

But let's not hurry with replacing these calls by hand: there's a [fixer rule for this](https://mlocati.github.io/php-cs-fixer-configurator/#version:2.16|fixer:native_function_invocation) that we can use.

### CODE REVIEW

File: tests/ExceptionTest.php [hunk 1]
```diff
@@ -85,78 +90,60 @@ public function testgetInlineParts()
...
-        $Parser = new Parser();
         $Parser->setStream('azerty');
     }
 
     /**
+     * @runInSeparateProcess
+     * @preserveGlobalState disabled
      */
-    public function testSetStreamWithoutTmpPermissions()
+    public function testGetAttachmentStreamWithoutTmpPermissions()
```
Comment (eXorus) [inline tests/ExceptionTest.php]: Here you don't test anymore the function getAttachments() but the setStream.
So the name of the test is wrong.

File: tests/ExceptionTest.php [hunk 2]
```diff
@@ -85,78 +90,60 @@ public function testgetInlineParts()
...
-     */
-    public function testGetAttachmentStreamWithoutTmpPermissions()
-    {
         $this->expectException(\PhpMimeMailParser\Exception::class);
         $this->expectExceptionMessage('Could not create temporary files for attachments.');
 
-
-        $file = __DIR__.'/mails/m0001';
-
         $Parser = new Parser();
```
Comment (eXorus) [inline tests/ExceptionTest.php]: This line should be before expectException like you did for the others tests.

File: tests/ExceptionTest.php (hunk 1 above)
Comment (sanmai) [inline tests/ExceptionTest.php]:
```suggestion
    public function testSetStreamWithoutTmpPermissions()
```

Comment (sanmai) [reply tests/ExceptionTest.php]: That's right, I missed this.

File: tests/ExceptionTest.php (hunk 2 above)
Comment (sanmai) [reply tests/ExceptionTest.php]: Corrected.

**Comment by eXorus** [comment]
Thanks again.

**Pull request merged**